analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -151,4 +155,11 @@ public override void ExecutionCompleted()
 
     private static bool IsEmptyArray(IOperation operation) =>
         operation.AsArrayCreation()?.DimensionSizes.Any(x => x.ConstantValue.Value is 0) ?? false;
+
+    private static ISymbol DictionaryWithAccessedSetter(IOperation operation) =>
+        operation.AsAssignment() is { } assignment
```

Comment:
If someone does `_ = dict[42]` than where 100% sure that it's not empty too.

So this should not be based on assignment, but rather the indexer property reference itself

Reply:
So that I understand correctly:
`_ = dict[42]`
If `dict` is empty, this will throw (and S6466 will raise).
Therefore we can assume that `dict` is not empty after this line.

Reply:
Exactly